### DISCUSSION TITLE
Possible problem with AMP?

### DISCUSSION THREAD

**Discussion by smitu** [vercel/next.js#15520] [Help]
We have getstaticprops loading data and we’re using dynamic routing.

Now we want to run AMP here, but the problem is that getstaticprops returns Undefinded. AMP sends Undefinded first and that makes it an error.

We can’t render the content because the AMP doesn’t have the data, only an empty object. Is there any options to fix it?

**Comment by ijjk** [comment]
Hi, can you provide a reproduction for the error you are running into? It sounds like you might be using `fallback: true` and not handling the [`fallback` case](https://nextjs.org/docs/basic-features/data-fetching#fallback-pages)

**Reply by smitu** [reply]
We have something like this: 

`function Post(props) {
   const isAmp = useAmp()
   return isAmp ? <SingleAmp data={props} /> : <PostPage data={props}/>
}

Post.getInitialProps = async ({ req, query }) => {
   return getPosts(serviceUrl+'/'+query.year+'/'+query.mounth+'/'+query.slug, { req, query })
}`

after change this is what we have: 

`function Post(props) {
  const router = useRouter()
  const isAmp = useAmp()
  const fetcher = (...args) => fetch(...args).then(res => res.json())
  const { data } = useSWR(serviceUrl+router.asPath, fetcher, { initialData: props.data })
  return isAmp ? <SingleAmp data={data}/> : <PostPage data={data}/>
}

export async function getStaticPaths() {
  return {
    paths: [{ params: {year: '2020', mounth: '07', slug: 'xyz'} }],
    fallback: true,
  }
}

export async function getStaticProps({ params }) {
  const res = await fetch(`${serviceUrl}/${params.year}/${params.mounth}/${params.slug}`)
  const data = await res.json()
  return { props: { data: data } }
}`

**Reply by ijjk** [reply]
So since you are using `fallback: true` you need to make sure to check `router.isFallback` and show a loading state since this page isn't generated during build time as mentioned [in the docs](https://nextjs.org/docs/basic-features/data-fetching#fallback-pages). This isn't specific to AMP pages and applies for non-AMP fallback pages as well

**Reply by smitu** [reply]
the problem is that I don't have a data to show anything in AMP 

data is only visible when the router.isFallback is on false and then only "loading" is visible in the AMP

**Reply by MistaPidaus** [reply]
@ijjk Getting similar issue. I already implement the `router.isFallback`, it stuck at "Loading" without change.
